geoportailv3/static/js/featurepopup/featurepopupservice.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @fileoverview Provides a feature popup service.
+ */
+
+goog.provide('app.FeaturePopup');
+
+goog.require('app.featurePopupDirective');
+goog.require('app.styleEditingDirective');
+goog.require('goog.asserts');
+goog.require('goog.dom');
+goog.require('ol.Coordinate');
+goog.require('ol.MapProperty');
+goog.require('ol.Overlay');
+
+
+
+/**
+ * @param {angular.$compile} $compile The compile provider.
+ * @param {angular.Scope} $rootScope The rootScope provider.
+ * @constructor
+ * @ngInject
+ */
+app.FeaturePopup = function($compile, $rootScope) {
+
+  /**
+   * @type {ol.Collection<ol.Feature>?}
+   * @private
+   */
+  this.features_;
+
+  /**
+   * The scope the compiled element is linked to.
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $rootScope.$new(true);
+
+  var element = angular.element('<div app-feature-popup></div>');
+  element.addClass('feature-popup');
+  element.attr('app-feature-popup-feature', 'feature');
+
+  // Compile the element, link it to the scope
+  $compile(element)(this.scope_);
+
+  /**
+   * @type {ol.Overlay?}
+   * @private
+   */
+  this.overlay_ = new ol.Overlay({
+    element: element[0],
+    autoPan: true,
+    autoPanAnimation: /** @type {olx.animation.PanOptions} */ ({
+      duration: 250
+    })
+  });
+};
+
+
+/**
+ * @param {ol.Map} map Map.
+ * @param {ol.Collection<ol.Feature>} features Features.
+ */
+app.FeaturePopup.prototype.init = function(map, features) {
+  map.addOverlay(this.overlay_);
+  this.features_ = features;
+};
+
+
+/**
+ * @param {ol.Feature} feature
+ * @param {ol.Coordinate=} opt_anchor
+ */
+app.FeaturePopup.prototype.show = function(feature, opt_anchor) {
+  this.scope_['feature'] = feature;
+  var anchor = goog.isDef(opt_anchor) ? opt_anchor : this.getAnchor(feature);
+  this.overlay_.setPosition(anchor);
+};
+
+
+/**
+ */
+app.FeaturePopup.prototype.hide = function() {
+  this.scope_['feature'] = null;
```

Comment:
Set `this['feature']` to `undefined. In this way you can use `goog.isDef` instead of `goog.isDefAndNotNull`.